test/auto-disco.cpp

```diff
@@ -71,72 +69,73 @@ TEST_CASE("Auto disco / readConfig", "[auto]")
+    CHECK(status.sts            == 0);
+    CHECK(status.sensors        == 0);
     discoAuto.updateStatusDiscoveryCounters("ups");
     discoAuto.updateStatusDiscoveryProgress();
     status = discoAuto.getStatus();
-    CHECK(status.ups        == 1);
-    CHECK(status.epdu       == 0);
-    CHECK(status.sts        == 0);
-    CHECK(status.sensors    == 0);
-    CHECK(status.discovered == 1);
-    CHECK(status.progress   == 25);
+    CHECK(status.addressScanned == 1);
+    CHECK(status.discovered     == 1);
+    CHECK(status.ups            == 1);
+    CHECK(status.epdu           == 0);
+    CHECK(status.sts            == 0);
+    CHECK(status.sensors        == 0);
     discoAuto.updateStatusDiscoveryCounters("epdu");
     discoAuto.updateStatusDiscoveryProgress();
     status = discoAuto.getStatus();
-    CHECK(status.ups        == 1);
-    CHECK(status.epdu       == 1);
-    CHECK(status.sts        == 0);
-    CHECK(status.sensors    == 0);
-    CHECK(status.discovered == 2);
-    CHECK(status.progress   == 50);
+    CHECK(status.addressScanned == 2);
+    CHECK(status.discovered     == 2);
+    CHECK(status.ups            == 1);
+    CHECK(status.epdu           == 1);
+    CHECK(status.sts            == 0);
+    CHECK(status.sensors        == 0);
     discoAuto.updateStatusDiscoveryCounters("sts");
     discoAuto.updateStatusDiscoveryProgress();
     status = discoAuto.getStatus();
-    CHECK(status.ups        == 1);
-    CHECK(status.epdu       == 1);
-    CHECK(status.sts        == 1);
-    CHECK(status.sensors    == 0);
-    CHECK(status.discovered == 3);
-    CHECK(status.progress   == 75);
+    CHECK(status.addressScanned == 3);
+    CHECK(status.discovered     == 3);
+    CHECK(status.ups            == 1);
+    CHECK(status.epdu           == 1);
+    CHECK(status.sts            == 1);
+    CHECK(status.sensors        == 0);
     discoAuto.updateStatusDiscoveryCounters("sensor");
     // caution normally no progress for sensor, just for test
     discoAuto.updateStatusDiscoveryProgress();
     status = discoAuto.getStatus();
-    CHECK(status.ups        == 1);
-    CHECK(status.epdu       == 1);
-    CHECK(status.sts        == 1);
-    CHECK(status.sensors    == 1);
-    CHECK(status.discovered == 4);
-    CHECK(status.progress   == 100);
+    CHECK(status.addressScanned == 4);
+    CHECK(status.discovered     == 4);
+    CHECK(status.ups            == 1);
+    CHECK(status.epdu           == 1);
+    CHECK(status.sts            == 1);
+    CHECK(status.sensors        == 1);
     // test limits
     discoAuto.updateStatusDiscoveryCounters("ups");
     discoAuto.updateStatusDiscoveryProgress();
     status = discoAuto.getStatus();
-    CHECK(status.ups        == 2);
-    CHECK(status.epdu       == 1);
-    CHECK(status.sts        == 1);
-    CHECK(status.sensors    == 1);
-    CHECK(status.discovered == 5);
-    CHECK(status.progress   == 100);
+    CHECK(status.addressScanned == 4);
+    CHECK(status.discovered     == 5);
```

Comment:
add a comment to explainwhy  discovered > addressScanned
about status struct:  add intermediate 'discovered' struct, as: discoved.total, discovered.ups, etc...

Reply:
Take into account